style updates and reduction of links in the navigation menu

charleslavon opened the pull request:
closes https://github.com/near/near-discovery/issues/955

calebjacob reviewed (changes requested):
Thanks for taking this on! Just noticed a couple minor issues.

1. The "Standards & Proposals" link is clipping through:

<img width="398" alt="Screen Shot 2024-02-21 at 3 48 10 PM" src="https://github.com/near/near-discovery/assets/1475067/87d4490e-98b9-4d97-b521-532980b5a2a3">

2. Hovering over an item adds a border to the icon (the icon should probably remain borderless):

<img width="345" alt="Screen Shot 2024-02-21 at 3 48 18 PM" src="https://github.com/near/near-discovery/assets/1475067/174cc707-b7ff-41b6-a853-2efb5ad28f80">

3. I wonder if it might look less cramped if we slightly increase the `gap` between the icon and text?

charleslavon commented:
good eye @calebjacob thanks for testing. I've fixed those issues. I fixed the alignment and hover issues by restructuring that row; one side-effect is that the nav option icons are not clickable, rather one must click on the name of the link.

calebjacob reviewed (approved):
The changes look good! The icon not being clickable is probably fine for now. If someone in the future wanted to refactor this, you could put the icon and text inside the anchor. Then you'd just wrap the text with a `<span>` and apply the underline to the nested `a > span` when the `a` is hovered.
